### PR TITLE
Enhance Test Coverage for Programme Commands

### PR DESCRIPTION
**Pull request by nirala-ts**
Parser Package: Added custom exception tests and enhanced edge case coverage for ParserUtils, FlagParser, CommandFactory, and ProgCommandFactory

Model Classes: Developed comprehensive tests for Day, Exercise, Programme, and ProgrammeList, covering key operations like addition, deletion, and retrieval.

Command Classes: Added tests for StartProgrammeCommand, ViewProgrammeCommand, and other related commands to verify correct functionality and handle invalid inputs.

### CODE REVIEW

File: src/main/java/programme/Exercise.java [hunk 1]
```diff
@@ -92,11 +90,10 @@ private void updateReps(Integer newReps) {
      * @param newWeight the new weight; if null, the weight is not updated
      */
     private void updateWeight(Integer newWeight) {
-        if (isNull(newWeight)) {
-            return;
+        if (newWeight != null) {
+            logger.log(Level.INFO, "Updating weight from {0} to {1}", new Object[]{weight, newWeight});
+            weight = newWeight;
```
Comment (TVageesan) [inline src/main/java/programme/Exercise.java:95]: It'd be best to preserve the [inversion pattern](https://www.jetbrains.com/help/resharper/InvertIf.html) here in order to reduce nesting.

File: src/main/java/programme/ExerciseUpdate.java [hunk 2]
```diff
@@ -11,10 +11,10 @@
  */
 public class ExerciseUpdate {
 
-    protected final int sets;
-    protected final int reps;
-    protected final int weight;
-    protected final int calories;
+    protected final Integer sets;
```
Comment (TVageesan) [inline src/main/java/programme/ExerciseUpdate.java:14]: using Integer instead of int + isNull() is a good idea, but it should be consistently applied across the codebase. isNull() is still used in FlagParser to check if argument string is null. Similarily, the use of NULL_INTEGER etc. constants should be ditched in places like ProgrammeList. Because of the scope of this change and the potential for introduction of bugs, I'd reccommend not doing it all.

**Review by TVageesan — APPROVED** [review]
Tests LGTM, but made some comments on changes you made to EditCommand
